README.md

```diff
@@ -21,32 +21,29 @@ is a common choice thanks to its natural concurrency model and portability.
 
 ## Library overview
 
-The library exposes the following packages:
-
-- `lattigo/he`: The main package of the library which provides scheme-agnostic interfaces
-  and Homomorphic Encryption for different plaintext domains.
-
-  - `hebin`: Homomorphic Encryption for binary arithmetic. It comprises blind rotations (a.k.a Lookup Tables) over RLWE ciphertexts.
-
-  - `hefloat`: Homomorphic Encryption for fixed-point approximate arithmetic over the complex or real numbers.
-
-    - `bootstrapping`: Bootstrapping for fixed-point approximate arithmetic over the real
-      and complex numbers, with support for the Conjugate Invariant ring, batch bootstrapping with automatic
-      packing/unpacking of sparsely packed/smaller ring degree ciphertexts, arbitrary precision bootstrapping,
-      and advanced circuit customization/parameterization.
-
-  - `heint`: Homomorphic Encryption for modular arithmetic over the integers.
+<p align="center" width="100%"">
+  <img width=500 height=350 alt="lattigo-hierarchy" src="./lattigo-hierarchy.svg">
+</p>
 
-- `lattigo/mhe`: Package for multiparty (a.k.a. distributed or threshold) key-generation and 
-  interactive ciphertext bootstrapping with secret-shared secret keys.
+Lattigo is a strictly hierarchical library whose packages form a linear dependency chain ranging from
+low-level arithmetic functionalities to high-level homomorphic circuits. A graphical depiction of the
+Lattigo package organization is given in the Figure above.
 
-  - `mhefloat`: Homomorphic decryption and re-encryption from and to Linear-Secret-Sharing-Shares, 
-    as well as interactive ciphertext bootstrapping for the package `he/hefloat`.
+- `lattigo/ring`: At the lowest level resides the `ring` package providing modular arithmetic operations for polynomials
+  in the RNS basis, including: RNS basis extension; RNS rescaling; number theoretic transform (NTT); uniform,
+  Gaussian and ternary sampling.
+  
+  - `lattigo/core`: This package implements the core cryptographic functionalities of the library and builds directly
+    upon the arithmetic functionalities provided by the `ring` package:
 
-  - `mheint`: Homomorphic decryption and re-encryption from and to Linear-Secret-Sharing-Shares, 
-    as well as interactive ciphertext bootstrapping for the package `he/heint`.
+	- `rlwe`: Common base for generic RLWE-based homomorphic encryption.
+      It provides all homomorphic functionalities and defines all structs that are not scheme-specific.
+      This includes plaintext, ciphertext, key-generation, encryption, decryption and key-switching, as
+      well as other more advanced primitives such as RLWE-repacking.
 
-- `lattigo/schemes`: A package implementing RLWE-based homomorphic encryption schemes.
+    - `rgsw`: A Full-RNS variant of Ring-GSW ciphertexts and the external product.
+	
+- `lattigo/schemes`: The implementation of RLWE-based homomorphic encryption schemes are found in the `schemes` package:
```

Comment:
This is an almost empty package, which duplicates the purpose of 'rlwe'. Should the two packages also be merged ? 

(As per my last email, but just putting this here to keep track).

Reply:
Which package are you referring to here?